### DISCUSSION TITLE
PLANS FOR FINDING REVENUE SOURCES FOR ASTROID

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#818] [Announcements]
Dear Astroiders,

It has been more than 3 months since I officially launched the Astroid crowdfunding page #705 . The amount of money raised from donors has reached $885 with 22 people joining in.

![Screen Shot 2024-09-26 at 15 27 16](https://github.com/user-attachments/assets/7b7723a4-9405-4942-a35b-a672a3ad0eed)

First of all, I would like to express my sincere thanks to all the supporters and volunteers from all over the world who have participated in this project and made Astroid stronger.

Second, I promise to try to maintain Astroid upgrades for as long as I continue to work on Joomla.

However, I have to admit that looking at the reality, the current amount of money collected from Astroid is not enough to support my current Joomla team. And it is terrible to realize that when I focus too much on Astroid, I have to ignore other TemPlaza projects like Jollyany. And recently when the revenue from Jollyany is decreasing significantly, I started thinking about trying to find more sources of income to continue maintaining this project. Because the Jollyany team told me that they will not be able to continue providing money to maintain the Astroid project.

That is why I am planning to create a commercial version for Astroid where there will be advanced features to help Astroid create a stable source of income enough to support itself without having to depend on anyone else.

### Here is a comparison of Astroid's plan
![Screen Shot 2024-09-26 at 13 53 29](https://github.com/user-attachments/assets/539af7eb-0210-440e-bbc2-4ebb2e816c0e)

Some of my commitments when developing the commercial version
1. Astroid's license will still be in the form of GPL
2. The current basic functions of Astroid will still be used as usual. Your website as well as your customers will not be affected by this upgrade.
3. You will still be able to use Astroid to develop your or your customers' websites. You can still use Astroid to develop templates and sell on your website.
4. The commercial version only includes the free version which integrates some more complex utilities and many features that bring better user experience such as GSAP or Dynamic Content etc.
5. And finally, I would like to pledge to all of you who have donated to Astroid and to all of you who have volunteered that you will get to use Astroid Pro for life for free. I hope this is my sincere thanks to all of you.

These are my thoughts on the plans for Astroid in the upcoming releases. Please let me know what you think about this.

Thanks & Best Regards,
Sonny

**Comment by vtheod** [comment]
Dear @sonvnn 
You are doing a great work and tend to transform Astroid to not just a framework but to a real great page builder.
Also, your commitment and effort not only to develop Astroid but also to support users as well as to adapt new features is extraordinary!
Keep up the good work and keep a business license for me ;)

**Reply by sonvnn** [reply]
Thank you very much. Everything is still in the planning stage. I will implement soon and announce to the community in stages ❤️

**Comment by deltapapa01** [comment]
Hi, 99 dollars a year for such a great product is perfectly fine. I just hope it stays that way and Astroid can support itself with it.
For example, I don't need a widget at all.
For me the layout builder and the design possibilities like colour, font, images and menu are most important.
In principle, a blank template with the above settings is sufficient for me, but this is certainly different for everyone.
I even find it a bit confusing with all the widgets, and all the updates are a bit much for me personally.
Maybe they should release new features and updates every few months. 
But like I said, $99 is totally fine!
You're doing a great job Sonny!

**Comment by Banuk1616** [comment]
@Sonny
When your call for donations came up, I wanted to throw the idea of offering a less extensive free version and a more extensive donor version into the room.

Basically, I am very disappointed that so few users have supported the Astroid Framework financially.

But unfortunately that's the way it is in this day and age - many people want to take something, but few want to give something back.

I support your idea, even though as an early retiree I don't have that much money and don't use the Astroid Framework commercially, but only use it to build websites for myself, a few friends and dear acquaintances for free. It's a hobby where I also learn something, but don't earn any money.
However, as deltapapa has already written, the updates are a bit too frequent for me too. Quick bug fixes yes, but updates with new functions don't need to be released so often in my opinion. 

Nevertheless, I have been pleased with every update with improvements and new functions so far and I really appreciate their work and commitment to the Astroid Framework.

As long as I continue to pursue this hobby, I will continue to use and support the Astroid Framework, it is simply the best.

Best regards

**Comment by paurray** [comment]
It goes without saying that Astroid is awesome and the work that @sonvnn and his team do is absolutely priceless. Thank you!

I love Joomla and I love Astroid even more because I really dig its integration with Joomla.
All at a very core level. In a bare bones kind of way. Meant in the best possible way.
It can be a bit twisted at times but if you go to the time to understand it, it makes soooooo much sense.

Completely understand the pricing model.
And that people need to be paid at the very, very, very latest when there is a commitment to support!
Things are tight here at the moment here. Sorry that my contribution was very much on the low end of the scale. 

I do how ever feel compelled to echo some of the comments from above:

a) Not wanting to steal any one’s thunder but the pace at which things develop around here sometimes feels insane. I have toyed with 100s of different types of software for a couple of decades now and I have never experienced anything like this. Maybe shift down a gear closer to marathon mode than sprint mode?

b) Also I have 3 thoughts about 80 +/- Widgets:

- I for one am more than happy with 5 +/- Widgets. Who really needs 80 ish Widgets? 

- Maybe have a system of voting for most desired widget? And implement in order of most requested?

- And finally developing 80 widgets is one thing, supporting people using 80 widgets and maintaining 80 widgets yet another thing?

I want to continue making tutorials which is another way of supporting the community but have a sick child at home all week :-(

Just my 2 cents.

Paul

**Comment by sonvnn** [comment]
Hi all,

@deltapapa01 @Banuk1616 Thanks for your helpful advice and suggestions. Usually, when software makes big strides, there will be upgrades close together. Sometimes we also see Joomla release 2 to 3 versions within a week. That only happens when a bug is discovered in the software that affects many people in the community, forcing the developer to release a patch. And you know, when Astroid went from 3.0 to 3.1 with Article Layout and Sub-Layout, it was no exception. I have received feedback about the bugs that have arisen and my priority is always to fix the problems immediately to avoid it affecting the community. When the system is stable, the release frequency will be 1-2 months for a feature upgrade version.

I see that Astroid 3.1 has reached a stable level at the moment. So we don't need to worry about the next problems :). 

@paurray Regarding the utility issue, it depends on each person's needs. Some people want less, but many people need more. For example, I have a list of features that need to be developed from @rinka88 LOL. In the near future, I will add a feature that allows you to hide and show the widgets you want. I believe it will solve the problem when there are too many Widgets coexisting.

Thanks & Best Regards,
Sonny

**Comment by Dnsftl** [comment]
@sonvnn 

I think it's a logical step to offer Astroid in a subscription model, especially considering the fact that there are currently few people donating. Personally, I can fully support the subscription options you're proposing; there are plenty of choices, and it provides a solid foundation to further support and grow the project. 

Astroid works excellently with Joomla; it's fast, reliable, and the support is outstanding. For me, frequent updates are not a problem—I'd rather have regular updates than wait too long for a solution. We will continue to actively support Astroid by keeping a permanent link to Astroid on our website.

Once again, thank you for all the hard work you've put into Astroid; it has become a fantastic product/software!

Regards,

Dennis

**Comment by Chacapamac** [comment]
$99 dollars for the full power of the best Joomla framework is nothing. @sonvnn , Count me as a pro user.

**Comment by joomlaplates-de** [comment]
Hi Sonny, I understand you and your situation. We always have to check where to get the money to live on. Here in Germany its not even better with selling templates ect. In my opinion it takes a while until the user realise "good products" cant be for free always.
Anyway, I would delete all widgets from Basic Astroid version and just keep the Astroid like it was in year 2022. I think Helix is doing the same way. Basic settings and if you want more, you have to pay.
And honestly, if the user has no 30 dollar the year then they should go somewhere else and find a product for free.
Wish you all the best, good luck, peace out. Peter

**Reply by rinka88** [reply]
widgets are a kind of builder. and free widgets are the free version of sp page builder. If you don’t need widgets, there is a disable button in the plugin settings.

**Reply by sonvnn** [reply]
@Joomlaplates The sad truth is that I am moving some parts to WordPress. I don’t want to because I have been with Joomla for over 15 years and I love the code. Performance-wise, Joomla 5 is 10 times faster than WordPress now. But we have lost too much resources, customer trust and market share to WordPress. Although WordPress is in a dispute with WP Engine. But beating them at this point is really impossible for Joomla. Their community is too strong and well-established.

Thank you for your cooperation!
Sonny

**Reply by joomlaplates-de** [reply]
In my opinion, Wordpress is completly useless for building webpages unless its a blog. The Wordpress code is terrible and the Editor is even more worst. Joomla is still the best, but selling Joomla themes isnt that easy anymore and on ThemeForest the most user buy Wordpress. I would offer Astroid on JDE and build a membership site for Astroid PRO with support forum and so on.
Same like Helix, basic version is for free and Pro version or Sitebuilder costs something.
Regards Peter

**Reply by jam1974** [reply]
I can only agree with Peter about WordPress. But it shouldn't be about the system that is only big because of marketing and not because of performance.
The Astroid framework with a template without any widgets should remain free because it already combines so many functions, especially in terms of customization and administration. Little basic knowledge is enough. It really doesn't have to be much. If you can't do that, if you need help, then pay $30 for a year for access to a manual and access to a support area. If you want to get more out of the template, you can buy the widgets for $30.  Selling point: everything from a single source. No problems with integration, no settings that have to be made twice. If you want the widgets and support for the widgets, you pay $60 and $90 for widgets, widgets support and template support.
However, I am also of the opinion that Joomla support should be chargeable. When I see posts in some forums where the answer is expected immediately, or you have to ask for parts first to be able to narrow down a issue. Or posts in which you are directly offered the access data so that you can quickly set it up the way the user wants it. Above all, if the support costs something, you could recognize the supporters and support the whole Joomla project. But well, we are in a time when something free should not cost anything.
But well, we are in a time in which something free should not cost anything. 
My opinion
Oliver

**Reply by Chacapamac** [reply]
Completely agree with @Joomlaplates — A version free without access to any widgets — You want the full version, You pay the fee. 

Also, many software today doesn’t give access to updates freely. This is normal as you should pay the developer time to keep the system secure and improve for you.

**Comment by joomlaplates-de** [comment]
Maybe another idea!!!
I would create a website > https://astroidframe.work/   and includes a support forum. And the customer has to be an active membership to post a qustion in forum. Because the most time I spend on support.
In fact, Astroid is free, but support they have to pay for.

**Reply by sonvnn** [reply]
I will try with Free/Pro model first. Then depending on the situation, I will continue to make further steps. I am trying to find a way to maintain Astroid as long as possible.

**Reply by joomlaplates-de** [reply]
Hi Sony, I would do it differently. Easy! Astroid is free and open source but for Support need a membership to access the forum or whatever!! Greatings Peter

**Reply by Chacapamac** [reply]
**No, no support means that the user will choose another product.** 
How do you test and know a system as complex as a framework is good without a minimum of support?
Or give a 3-month free support. After, you need to take a subscription. 3 months is enough to have a good idea of the quality of the product.

**Comment by iorbita** [comment]
Sorry I haven't been very active in these discussions. I'm currently finalizing a long-term project (with Astroid, of course!) and haven't had much time to participate. However, I've been following the conversations closely.

Like many other Astroid enthusiasts, I'm thrilled about the upcoming Pro version. Sonny, I want to express my sincere gratitude for your hard work in reviving Astroid.

While I'm not sure about the priority of widgets, I think it's an interesting idea. Instead of installing numerous extensions, having integrated widgets within the framework could streamline the process and reduce the need for additional installations.

Most importantly, I hope Astroid continues to receive regular updates, aligned with Joomla's release schedule.

And in anyway, as you can imagine, I'm all for a Pro version. 🚀

Lorenzo

**Reply by joomlaplates-de** [reply]
There is only one problem with the widgets, ofc I like them BUT you cant assign them like a module position.
Ofc you can assign them, if you create different Template styles and then assign th styles to menu item >> but this is completly nonsense... Imagine you want to change a color, you have to change the color on all Template styles...

**Reply by iorbita** [reply]
... right, this remains a constraint...

**Reply by sonvnn** [reply]
@Joomlaplates @iorbita Thanks for bringing this up. I was also thinking about creating a module where you can put the Layout Builder. So you can load the Widgets inside the module. That would solve this problem.

**Reply by Banuk1616** [reply]
@sonvnn 
That would be brilliant.

**Reply by rinka88** [reply]
if you add module selection in widgets, now only the position is selected, this is already great

**Reply by kvintvs** [reply]
> @Joomlaplates @iorbita Thanks for bringing this up. I was also thinking about creating a module where you can put the Layout Builder. So you can load the Widgets inside the module. That would solve this problem.

It would be wonderful to have a way to load and edit widgets inside modules. For multilingual websites, that'd mean that widgets can be translated like any other piece of Joomla content such as articles and modules, so we can avoid duplicated template styles. This would also solve the incompatibility problem with the translation plugin Falang I guess, as we would simply translate the widget modules.

**Comment by srcarney** [comment]
I like the idea of the Business Pro plan for $99 per year. Seems like a bargain to me and I sure do want to see Astroid succeed. Quite a few people commented that there were too many updates. But those updates to me are a feature and not a bug. It shows you are in active development which I really appreciate.

**Comment by sonvnn** [comment]
Thank you for your feedback and support of Astroid. After our team meeting earlier this week, based on our current financial situation and comparisons with our competitors, we have agreed to the following pricing for the Pro package:

- Essential: $59/year
- Business: $99/year
- Elite: $199/year

You can see the details directly [here](https://astroidframe.work/pricing)

With my best regards,
Sonny

**Comment by srcarney** [comment]
I ordered Essentials for now.  Somehow a brand new account was created when
I ordered. I have a lot of confusion whenever I order from template because
my login and password seems to be disregarded and a new account gets
created. Seems really confusing and it makes it hard for me to keep track
of my purchases. I wish I could only have one account and make all my
purchase through that one account. Instead I have multiple logins being
created. I have purchased a lifetime template license and an Astroid
Essentials license so far. I will probably get a portfolio licence
eventually and hope I can keep them all together under one login.

**Reply by sonvnn** [reply]
> I ordered Essentials for now. Somehow a brand new account was created when I ordered. I have a lot of confusion whenever I order from template because my login and password seems to be disregarded and a new account gets created. Seems really confusing and it makes it hard for me to keep track of my purchases. I wish I could only have one account and make all my purchase through that one account. Instead I have multiple logins being created. I have purchased a lifetime template license and an Astroid Essentials license so far. I will probably get a portfolio licence eventually and hope I can keep them all together under one login.

Thank you very much for supporting Astroid.
Our system identifies buyers via email. Just use the same email for your next purchases. The licenses will be automatically collected into the same account without creating any new accounts :)

**Reply by iorbita** [reply]
Hi, same for me, for the moment I've switched to "Essential" plan because I'm only using Astroid for a personal project, but I won't hesitate to upgrade to "Business" version as soon as I have other projects to work on.

**Comment by srcarney** [comment]
Awesome, thank you so much!  I am very happy to support Astroid. I am sure
I will be upgrading my license in the future.

Sean

On Fri, Nov 22, 2024 at 8:00 PM Sonny Le ***@***.***> wrote:

> I ordered Essentials for now. Somehow a brand new account was created when
> I ordered. I have a lot of confusion whenever I order from template because
> my login and password seems to be disregarded and a new account gets
> created. Seems really confusing and it makes it hard for me to keep track
> of my purchases. I wish I could only have one account and make all my
> purchase through that one account. Instead I have multiple logins being
> created. I have purchased a lifetime template license and an Astroid
> Essentials license so far. I will probably get a portfolio licence
> eventually and hope I can keep them all together under one login.
>
> Thank you very much for supporting Astroid.
> Our system identifies buyers via email. Just use the same email for your
> next purchases. The licenses will be automatically collected into the same
> account without creating any new accounts :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/templaza/astroid-framework/discussions/818#discussioncomment-11354595>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4PRKUV6PQ3LMFR37XKRZT2B7OSLAVCNFSM6AAAAABO4Q5J7CVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMZVGQ2TSNI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 

Faith - Family - Friends - Fun.
Take time for all four. Then back to work!
